src/AppInstallerCommonCore/Filesystem.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "AppInstallerProgress.h"
+#include "winget/Registry.h"
+
+namespace AppInstaller::Filesystem
+{
+    namespace
+    {
+        DWORD CALLBACK CopyFileProgressCallback(
+            LARGE_INTEGER TotalFileSize,
+            LARGE_INTEGER TotalBytesTransferred,
+            LARGE_INTEGER StreamSize,
+            LARGE_INTEGER StreamBytesTransferred,
+            DWORD dwStreamNumber,
+            DWORD dwCallbackReason,
+            HANDLE hSourceFile,
+            HANDLE hDestinationFile,
+            LPVOID lpData
+        )
+        {
+            UNREFERENCED_PARAMETER(StreamSize);
+            UNREFERENCED_PARAMETER(StreamBytesTransferred);
+            UNREFERENCED_PARAMETER(hSourceFile);
+            UNREFERENCED_PARAMETER(hDestinationFile);
+
+            ProgressCallback callback = static_cast<ProgressCallback>((ProgressCallback*)lpData);
+
+            if (callback.IsCancelled())
+            {
+                return PROGRESS_CANCEL;
+            }
+
+            if (dwCallbackReason == CALLBACK_STREAM_SWITCH || dwStreamNumber == 1)
+            {
+                callback.BeginProgress();
+            }
+
+            if (dwCallbackReason == CALLBACK_CHUNK_FINISHED)
+            {
+                callback.OnProgress(TotalBytesTransferred.QuadPart, TotalFileSize.QuadPart, AppInstaller::ProgressType::Percent);
+            }
+
+            if (TotalBytesTransferred.QuadPart == TotalFileSize.QuadPart)
+            {
+                callback.EndProgress(true);
+            }
+
+            return PROGRESS_CONTINUE;
+        }
+    }
+
+    void CreateSymlink(const std::filesystem::path& target, const std::filesystem::path& link)
+    {
+        if (std::filesystem::exists(link))

```

Comment:
I removed the helper function and handle it explicitly in the main portable install flow. 

My logic for testing if the link is as follows:
1. Check if file is a symlink by using std::fs::is_symlink(). If it is a symlink file, we remove it.
2. Check if file still exists and remove it if it does to avoid collision. 

This should now work as expected since we have already verified whether the file is a symlink file to avoid any confusion of std::fs::exists checking the target of the symlink. We can now call std::fs::exists on the non-link file if one still exists and remove it.